Architecture/TodoApp/android/core/src/main/java/com/novoda/todoapp/tasks/service/PersistedTasksService.java

```diff
@@ -101,46 +121,30 @@ public Tasks apply(Tasks input) {
         };
     }
 
-    private Observable<Event<Tasks>> initialiseSubject() {
-        return Observable.defer(new Func0<Observable<Event<Tasks>>>() {
-            @Override
-            public Observable<Event<Tasks>> call() {
-                if (isInitialised(taskRelay)) {
-                    return Observable.empty();
-                }
-                return localDataSource.getTasks()
-                        .flatMap(fetchFromRemoteIfOutOfDate())
-                        .switchIfEmpty(fetchFromRemote())
-                        .compose(asValidatedEvent(taskRelay.getValue()))
-                        .doOnNext(taskRelay)
-                        .publish().autoConnect();
-            }
-        });
-    }
-
-    private Func1<Tasks, Observable<Tasks>> fetchFromRemoteIfOutOfDate() {
+    private Func1<Tasks, Observable<Tasks>> fetchFromRemoteIfOutOfDate(final long actionTimeInMillis) {
         return new Func1<Tasks, Observable<Tasks>>() {
             @Override
             public Observable<Tasks> call(Tasks tasks) {
                 if (tasksDataFreshnessChecker.isFresh(tasks)) {
                     return Observable.just(tasks);
```

Comment:
Just for my curiosity, could you not have a filter on `localDataSource.getTasks()` instead of using an Observable.just? I guess you would have an extra couple of actions to perform in your pipeline.